core/src/main/java/org/projectnessie/cel/extension/StringsLib.java

```diff
@@ -297,14 +422,147 @@ static String replace(String str, String old, String replacement) {
     return str.replace(old, replacement);
   }
 
+  /**
+   * replace first n non-overlapping instance of {old} replaced by {replacement}. It works as
+   * https://pkg.go.dev/strings#Replace to have consistent behavior as cel in Go
+   *
+   * <p>if n == 0, there is no change to the string
```

Comment:
Can you surround the code-like snippets like `n == 0` with `{@code ...}` here and in the other javadoc comments?